pkg/proxy/endpointslicecache.go

```diff
@@ -119,12 +123,17 @@ func newEndpointSliceInfo(endpointSlice *discovery.EndpointSlice, remove bool) *
 
 	if !remove {
 		for _, endpoint := range endpointSlice.Endpoints {
-			if endpoint.Conditions.Ready == nil || *endpoint.Conditions.Ready {
-				esInfo.Endpoints = append(esInfo.Endpoints, &endpointInfo{
-					Addresses: endpoint.Addresses,
-					Topology:  endpoint.Topology,
-				})
+			epInfo := &endpointInfo{
+				Addresses: endpoint.Addresses,
+				Topology:  endpoint.Topology,
+
+				// conditions
+				Ready:       endpoint.Conditions.Ready == nil || *endpoint.Conditions.Ready,
+				Serving:     endpoint.Conditions.Serving == nil || *endpoint.Conditions.Serving,
```

Comment:
I had assumed it should because this is the case for `Ready`.

Reply:
maybe if `serving == nil` we check ready, but that would mean that we could never have ready terminating endpoints if serving is nil, maybe that's fine since epslice controller will always set them in lock step starting v1.21.

Reply:
yeah I think falling back to ready might be a bit better here but agree that it's not obvious which is better. Either way a comment might help explain the logic here since there is some ambiguity.

Reply:
Alternatively what we're doing with `terminating` here (false if not specified) could also potentially be reasonable since we're primarily relying on `ready` still.

Reply:
These two statements are not aligned with the doc you have here: https://github.com/kubernetes/kubernetes/pull/96371/files#diff-e98883cc247c5cfc2f74946972a17e2b381ba288b35421db7bc9776612eb0366R54

Reply:
Yeah, that's a good point. Maybe the issue is that we're trying to translate 3 possible values (`*bool`) to 2 (`bool`) and it's not entirely clear what the nil case should be. It makes sense to default `terminating` to false when it isn't specified, I'm not sure that there is a sane default when `serving` isn't specified. Can we use a `*bool` for this instead?